### PR TITLE
Update local development instructions

### PR DESCRIPTION
**Pull request by judemars**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design

/kind documentation

> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Updates local development documentation with more detail + adds new needed role for e2e testing.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/local-development.md [hunk 1]
```diff
@@ -3,19 +3,36 @@ This page contains information on how to develop and test the driver locally.
...
+$ export IAM_NAME=$GCE_PD_SA_NAME@$PROJECT.iam.gserviceaccount.com  # IAM SA that was set up in "one time setup"
+
 $ ./test/run-e2e-local.sh
 ```
 
-Running Sanity Tests:
+### Sanity Tests
+> **_NOTE:_**  Sanity tests are currently failing, tracked by go/pdcsi-oss-driver/issues/990.
+
+Unit tests can be run from VS Code, etc directly or via the cmd line:
```
Comment (mattcary) [inline docs/local-development.md:29]: s/Unit/Sanity/?

Comment (judemars) [reply docs/local-development.md]: Woops, fixed! Thanks

**Comment by mauriciopoppe** [comment]
There's also https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/docs/kubernetes/development.md

/ok-to-test

**Comment by judemars** [comment]
@mauriciopoppe is the request to combine them?

File: docs/local-development.md [hunk 2]
```diff
@@ -3,19 +3,36 @@ This page contains information on how to develop and test the driver locally.
...
+$ export PROJECT=my-project                               # GCP Project to run tests in
+$ export GCE_PD_SA_NAME=$PROJECT-pd-sa
+$ export IAM_NAME=$GCE_PD_SA_NAME@$PROJECT.iam.gserviceaccount.com  # IAM SA that was set up in "one time setup"
+
 $ ./test/run-e2e-local.sh
 ```
 
-Running Sanity Tests:
+### Sanity Tests
+> **_NOTE:_**  Sanity tests are currently failing, tracked by go/pdcsi-oss-driver/issues/990.
```
Comment (mauriciopoppe) [inline docs/local-development.md:27]: This link only works internally

**Review by mauriciopoppe — COMMENTED** [review]
Just wanted to point out that there is another set of docs to install the driver on an existing k8s cluster. I haven't read what `./test/run-e2e-local.sh` but I usually install the driver and then run the e2e tests as different steps

File: docs/local-development.md (hunk 2 above)
Comment (judemars) [reply docs/local-development.md]: Great point, fixed! Thanks

**Comment by judemars** [comment]
Alrighty @mauriciopoppe I combined the guides, I think it makes sense to have the testing instructions after the installation instructions. Lmk what you think

**Pull request merged**